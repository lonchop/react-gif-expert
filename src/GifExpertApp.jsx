import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
import { v4 as uuidv4 } from 'uuid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch'])

  const onAddCategory = (newCategory) => {
    setCategories([newCategory, ...categories])
  }

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid
          key={uuidv4()}
          category={category}
        />
      ))}
    </>
  )
}
