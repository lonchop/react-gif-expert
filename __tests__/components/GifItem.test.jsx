import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components'

/* eslint-disable no-undef */
describe('Pruebas en <GifItem />', () => {
  const title = 'Soy un titulo'
  const url = 'http://localhost:5173/'
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(
      <GifItem
        title={title}
        url={url}
      />
    )
    expect(container).toMatchSnapshot()
  })

  test('Debe mostrar la imagen con el URL y el ALT indicados', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    )
    // screen.debug()

    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)
    // Es igual que lo anterior, pero de manera mas corta y mejor
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Debe de mostrar el titulo en el componente', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    )

    expect(screen.getByText(title)).toBeTruthy()
  })
})
