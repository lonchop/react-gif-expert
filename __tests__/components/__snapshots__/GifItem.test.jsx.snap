// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un titulo"
      src="http://localhost:5173/"
    />
    <p>
      Soy un titulo
    </p>
  </div>
</div>
`;

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Soy un titulo"
      src="http://localhost:5173/"
    />
    <p>
      Soy un titulo
    </p>
  </div>
</div>
`;
